### ISSUE TITLE
System.Random.MWC.Distributions.categorical: bad weights!

### ISSUE THREAD

**Issue by ghost** [kaizhang/clustering#5]
I get this error with random and default seed.

**Comment by kaizhang** [comment]
Do you have the code so that I can reproduce the error?

**Comment by ghost** [comment]
Read the [matrix](http://klacansky.com/data/paths.txt) as `Matrix Vector.Unboxed Double`  and run it through following command:
```
kmeans 2 normalized defaultKMeansOpts { kmeansMethod = KMeansPP, kmeansClusters = False } 
```

**Comment by kaizhang** [comment]
The reason is that your input matrix contains `NaN` values. For example, 

```
> print $ (reverse $ MU.toRows inputMatrix) !! 3
[1.932910689362377e-2,NaN,3.735307147459916e-2,NaN,1.2075905692926834e-2]
```

**Comment by kaizhang** [comment]
fix in 09e937c3750055b5f62cef3204f5e73c77904f38

**Comment by ghost** [comment]
Thanks, I did not notice `NaN`s.
